clients/banking/src/components/TransferInternationalWizardBeneficiary.tsx

```diff
@@ -60,159 +58,166 @@ export const TransferInternationalWizardBeneficiary = ({
     name: { initialValue: initialBeneficiary?.name ?? "", validate: validateRequired },
   });
 
-  const handleOnRouteChange = useCallback((route: string) => {
-    setRoute(route);
-  }, []);
-
   const handleOnResultsChange = useDebounce<ResultItem[]>(value => {
-    setResults(value.filter(({ value }) => isNotEmpty(value)));
+    const nextResults = value.filter(({ value }) => isNotEmpty(value));
+    setResults(nextResults);
+    setVariables({ dynamicFields: nextResults });
   }, 1000);
 
-  return (
-    <View>
-      <Tile
-        footer={match(errors)
-          .with(P.union([], P.nullish), () => null)
-          .with([P.select()], error => <LakeAlert anchored={true} variant="error" title={error} />)
-          .otherwise(errors => (
-            <LakeAlert
-              anchored={true}
-              variant="error"
-              title={t("transfer.new.internationalTransfer.errors.title")}
-            >
-              {errors.map((message, index) => (
-                <LakeText key={`validation-alert-${index}`}>{message}</LakeText>
-              ))}
-            </LakeAlert>
-          ))}
-      >
-        <LakeLabel
-          label={t("transfer.new.internationalTransfer.beneficiary.name")}
-          render={id => (
-            <Field name="name">
-              {({ value, onChange, onBlur, error, valid, ref }) => (
-                <LakeTextInput
-                  ref={ref}
-                  id={id}
-                  value={value}
-                  error={error}
-                  valid={valid}
-                  onChangeText={onChange}
-                  onBlur={onBlur}
-                  inputMode="text"
-                />
-              )}
-            </Field>
-          )}
-        />
+  return match(data)
+    .with(AsyncData.P.NotAsked, AsyncData.P.Loading, () => (
+      <ActivityIndicator color={colors.gray[500]} />
+    ))
+    .with(AsyncData.P.Done(Result.P.Error(P.select())), error => <ErrorView error={error} />)
+    .with(AsyncData.P.Done(Result.P.Ok(P.select())), ({ internationalBeneficiaryDynamicForms }) => {
+      if (internationalBeneficiaryDynamicForms == null) {
+        return <ErrorView />;
+      }
 
-        <Space height={8} />
+      const schemes = internationalBeneficiaryDynamicForms.schemes;
 
-        {data.isLoading() ? (
-          <ActivityIndicator color={colors.gray[500]} />
-        ) : (
-          <TransitionView {...(reloadingData.isLoading() && animations.heartbeat)}>
+      const routes = schemes.map(({ type }) => ({
+        value: type,
+        name: getInternationalTransferFormRouteLabel(type),
+      }));
+
+      const firstRoute = routes.at(0);
+
+      if (firstRoute == null) {
+        return <ErrorView />;
+      }
+
+      const selectedRoute = route.getOr(firstRoute.value);
+
+      const fields = Array.findMap(schemes, scheme =>
+        scheme.type === selectedRoute ? Option.Some(scheme.fields) : Option.None(),
+      ).getOr([]);
```

Comment:
nit, but I find

```ts
const fields = Array.find(schemes, scheme => scheme.type === selectedRoute)
  .map(scheme => scheme.fields)
  .getOr([]);
```

easier to understand (and no need to create `Option` instance on iteration)